content/courses/Ada_For_The_C_Embedded_Developer/chapters/04_Embedded.rst

```diff
@@ -231,9 +231,209 @@ Understanding Bare-Metal Environment
 Dealing with Absence of FPU with Fixed Point
 --------------------------------------------
 
-.. todo::
+Many numerical applications typically use floating-point types to compute
+values. However, in some platforms, a floating-point unit may not be available.
+Other platforms may have a floating-point unit, but using it in certain
+numerical algorithms can be prohibitive in terms of performance. For those
+cases, fixed-point arithmetic can be a good alternative.
 
-    Complete section!
+The difference between fixed-point and floating-point types might not be so
+obvious when looking at this code snippet:
+
+[Ada]
+
+.. code-block:: ada
+
+    --  [...]
+
+       Float_Value : Float := 0.25;
+       Fixed_Value : Fixed := 0.25;
+
+    begin
+
+       Float_Value := Float_Value + 0.25;
+       Fixed_Value := Fixed_Value + 0.25;
+
+       Put_Line ("Float_Value = " & Float'Image (Float_Value));
+       Put_Line ("Fixed_Value = " & Fixed'Image (Fixed_Value));
+
+    --  [...]
+
+In this example, the application will show the value 0.5 for both
```

Comment:
0.25 surely?

Reply:
I'm a little bit lost here. Do you mean that the application should show 0.25 instead of 0.5? It should actually be 0.5, since we're initializing both variables with 0.25 and then adding up 0.25, which gives us 0.5.

In any case, I've extended the example and added a run button (so that readers that check the result as well).